undertow/src/main/java/org/wildfly/extension/undertow/HostDefinition.java

```diff
@@ -86,8 +88,13 @@ public void marshallAsAttribute(AttributeDefinition attribute, ModelNode resourc
             .setDefaultValue(new ModelNode("ROOT.war"))
             .build();
 
+    static final SimpleAttributeDefinition DEFAULT_RESPONSE_CODE = new SimpleAttributeDefinitionBuilder(Constants.DEFAULT_RESPONSE_CODE, ModelType.INT, true)
+    .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+    .setValidator(new IntRangeValidator(400, 599, true, true))
```

Comment:
Should this have .setDefaultValue(new ModelNode(404)) ?

That's the basic behavior, and declaring it helps document the attribute.